src/language-js/clean.js

```diff
@@ -89,6 +89,11 @@ function clean(ast, newObj, parent) {
     delete newObj.key;
   }
 
+  if (ast.type === "OptionalMemberExpression" && ast.optional === false) {
+    newObj.type = "MemberExpression";
+    delete newObj.optional;
+  }
```

Comment:
@existentialism you may know what's going on here

```js
(a?.b).c
a?.b.c
```

give slightly different ast. In the second one, the outer `.` is

```js
{
  type: "OptionalMemberExpression",
  optional: false,
}
```

This doesn't make sense to me, why doesn't it print `{type: "MemberExpression"}` instead?

Reply:
Think @jridgewell is going to explain it better than I can, but AFAIK it's due to https://github.com/tc39/proposal-optional-chaining#edge-case-grouping

Reply:
This is intentional. The first example is explicitly breaking the optional chain by adding parenthesis. Imagine it were a sequence instead: `(0, a?.b).c`. Obviously in this case, I'm doing a regular property lookup on the result of the sequence.

The second case, however, is a continued optional chain. This shown by the ouptut: `a == null ? undefined : a.b.c`. The `.c` property access is "optional" in that it can be short-circuited. It's just not hard optional like the `a?.b` is.

Reply:
Is it safe (won't change behavior) to remove parenthesis there?

Reply:
Removing the parens changes the runtime chaining behavior, so you wouldn't want to remove them.

Reply:
No, they have different behavior: http://babeljs.io/repl/build/master#?babili=false&browsers=ie%2010&build=&builtIns=usage&code_lz=DYUwLgBAHgjBC8ECGB-AdAIzQYwFCkigCYEIAKVTAShyA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&lineWrap=true&presets=es2015%2Cstage-1&prettier=false&targets=&version=7.0.0-beta.48&envVersion=

However, there is no reason to every have parenthesis around an optional chain. The only observable difference between the `a?.b.c` and `a.b.c` expression (without the optional chain) is the error message. If you wanted it to throw, just use regular property chaining. If you wanted it to short-circuit, then you definitely didn't want it to throw.

Reply:
Thanks! I just opened an issue to track it: #4538. We didn't use to put parenthesis even before this update.